### PR TITLE
Fix the bug that hack/update-vendor.sh exits with error but no indication of failure

### LINKED ISSUE #99461: hack/update-vendor.sh can exit with error code 1 and no indication of failure
#### What happened:

Seen while running on https://github.com/kubernetes/kubernetes/pull/98572

```
$ hack/update-vendor.sh 
+++ [0225 09:14:27] logfile at /tmp/update-vendor.rLjX/update-vendor.log
+++ [0225 09:14:27] go.mod: update staging references
$ echo $?
1
$ tail /tmp/update-vendor.rLjX/update-vendor.log
warning: ignoring symlink /Users/liggitt/go/src/k8s.io/kubernetes/cluster/gce/ubuntu
warning: ignoring symlink /Users/liggitt/go/src/k8s.io/kubernetes/third_party/etcd
go: finding module for package google.golang.org/grpc/naming
k8s.io/kubernetes/pkg/registry/registrytest imports
	k8s.io/apiserver/pkg/storage/etcd3/testing imports
	go.etcd.io/etcd/integration imports
	go.etcd.io/etcd/proxy/grpcproxy imports
	google.golang.org/grpc/naming: module google.golang.org/grpc@latest found (v1.36.0, replaced by google.golang.org/grpc@v1.31.0), but does not contain package google.golang.org/grpc/naming
```

#### What you expected to happen:

If hack/update-vendor.sh fails, the output should make it clear

#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

/area code-organization
/sig architecture
/cc @justaugustus 
/assign

### PR DESCRIPTION
**Pull request by SataQiu**
#### What type of PR is this?
/kind bug
/kind cleanup

#### What this PR does / why we need it:
Fix the bug that hack/update-vendor.sh exits with error but no indication of failure

#### Which issue(s) this PR fixes:

Fixes #99461

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by jiahuif** [comment]
Looks like tests not running. See if `/ok-to-test` triggers tests.
/ok-to-test

**Comment by BenTheElder** [comment]
that was https://github.com/kubernetes/test-infra/issues/21090
`/test all` is a good option to try triggering tests when none are triggering

**Comment by SataQiu** [comment]
Hi, @fejta Could you please take a look at this PR?

File: hack/update-vendor.sh [hunk 1]
```diff
@@ -44,6 +44,15 @@ TMP_DIR="${TMP_DIR:-$(mktemp -d /tmp/update-vendor.XXXX)}"
...
 
+function finish {
+  ret=$?
+  if [[ ${ret} != 0 ]]; then
+    echo "An error has occurred. Please see more details in ${LOG_FILE}"
+  fi
+  exit ${ret}
+}
+trap finish EXIT
+
```
Comment (liggitt) [inline hack/update-vendor.sh:55]: cc @cblecker or @BenTheElder or other bash-savvy folks to ack this approach

**Review by fejta — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**